In [1]:
## Import packages 

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statistics
import seaborn as sns
%matplotlib inline

## Load File

df = pd.read_csv('https://raw.githubusercontent.com/borja876/Thinkful-DataScience-Borja/master/WELLCOME_APCspend2013_forThinkful.csv', encoding='latin1')

In [60]:
## View DataFrame
df

PMID/PMCID  \
0                                  NaN   
1                           PMC3679557   
2                23043264  PMC3506128    
3                  23438330 PMC3646402   
4                 23438216 PMC3601604    
5                           PMC3579457   
6                           PMC3709265   
7                 23057412 PMC3495574    
8                    PMCID: PMC3780468   
9                    PMCID: PMC3621575   
10                   PMCID: PMC3739413   
11                   PMCID: PMC3530961   
12                   PMCID: PMC3624797   
13                          PMC3413243   
14                          PMC3694353   
15                          PMC3572711   
16                            22610094   
17                   PMCID: PMC3586974   
18        23455506  PMCID: PMC3607399    
19          PMID: 24015914 PMC3833349    
20                       : PMC3805332    
21                                 NaN   
22            PMCID:\n    PMC3656742\n   
23                      PMCID: 3584654   
24                 22971149 PMC3466778   
25                            23072351   
26                          PMC3477822   
27                          PMC3530960   
28                23458425 PMC3630740    
29                      PMID: 20146481   
...                                ...   
2097                           3575693   
2098        PMID: 22188548 PMC3492912    
2099        PMID: 22764734 PMC3798095    
2100                    PMID: 23703895   
2101                 PMCID: PMC3664106   
2102                    PMID: 23649934   
2103                        PMC2955965   
2104        PMID: 19948006 PMC3551259    
2105                    PMID: 23305527   
2106                 PMCID: PMC3600532   
2107                        PMC3440570   
2108              23670821 PMC3738939    
2109                          23640856   
2110                    PMID: 23918316   
2111                 PMCID: PMC3728731   
2112       23495205685.88 PMC3743159     
2113         PMID:22997008 PMC3569869    
2114                           3708104   
2115                        PMC3713766   
2116                        PMC3819359   
2117                       PMC3773237    
2118                       PMC3815011    
2119                        PMC3805356   
2120                             EPUB    
2121                        PMC3707567   
2122                           2901593   
2123                           3748854   
2124                           3785148   
2125          PMCID:\n    PMC3647051\n   
2126  PMID: 23846567  (EPUB JULY 2013)   

                                         Publisher  \
0                                              CUP   
1                                              ACS   
2                                              ACS   
3                                              ACS   
4                                              ACS   
5                                              ACS   
6                                              ACS   
7                                              ACS   
8     ACS (AMERCIAN CHEMICAL SOCIETY) PUBLICATIONS   
9     ACS (AMERCIAN CHEMICAL SOCIETY) PUBLICATIONS   
10    ACS (AMERCIAN CHEMICAL SOCIETY) PUBLICATIONS   
11    ACS (AMERCIAN CHEMICAL SOCIETY) PUBLICATIONS   
12    ACS (AMERCIAN CHEMICAL SOCIETY) PUBLICATIONS   
13                                ACS PUBLICATIONS   
14                                ACS PUBLICATIONS   
15                                   AGA INSTITUTE   
16                                           AMBSB   
17           AMERICAN ASSOCIATION OF IMMUNOLOGISTS   
18           AMERICAN ASSOCIATION OF IMMUNOLOGISTS   
19                       AMERICAN CHEMICAL SOCIETY   
20                       AMERICAN CHEMICAL SOCIETY   
21                       AMERICAN CHEMICAL SOCIETY   
22                       AMERICAN CHEMICAL SOCIETY   
23                       AMERICAN CHEMICAL SOCIETY   
24                       AMERICAN CHEMICAL SOCIETY   
25                 

## Determine the five most common journals and the total articles for each

In [78]:
## View counts of each Journal Entry (as typed)
print(df['Journal title'].value_counts())

PLOS ONE                                                            210
JOURNAL OF BIOLOGICAL CHEMISTRY                                      53
NEUROIMAGE                                                           29
PLOS GENETICS                                                        24
PLOS PATHOGENS                                                       24
NUCLEIC ACIDS RESEARCH                                               23
PLOS NEGLECTED TROPICAL DISEASES                                     20
PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES                      20
HUMAN MOLECULAR GENETICS                                             19
NATURE COMMUNICATIONS                                                19
BMC PUBLIC HEALTH                                                    14
BRAIN                                                                14
MOVEMENT DISORDERS                                                   13
JOURNAL OF NEUROSCIENCE                                         

In [77]:
df.sort_values('Journal title')

PMID/PMCID  \
439                      PMC3378987\n   
21                                NaN   
20                      : PMC3805332    
19         PMID: 24015914 PMC3833349    
9                   PMCID: PMC3621575   
8                   PMCID: PMC3780468   
22           PMCID:\n    PMC3656742\n   
34                           23373658   
23                     PMCID: 3584654   
927          PMCID:\n    PMC3727331\n   
922         PMID:22993091 PMC3447403    
921                  \n    PMC3498934   
920                 PMCID: PMC3565438   
923                        PMC3087623   
928                 PMCID: PMC3668577   
929                 PMCID: PMC3606566   
1711                      PMC3808818    
1716  PMID: 23828613 (JULY 2013 EPUB)   
924                        PMC3374517   
1717                       PMC3549237   
1719                       PMC3661931   
1720                          3535376   
1718                              NaN   
2051                       PMC3798121   
1871     21624095  PMCID: PMC3734623    
1872                       IN PROCESS   
1873                         23734913   
1721                              NaN   
1722                              NaN   
1723                              NaN   
...                               ...   
954                        PMC3717178   
2018                       PMC3627817   
2006                       PMC3770928   
2005                              NaN   
2004                       PMC3558801   
955                        PMC3775257   
817            PMCID:\n    PMC3608034   
816            PMCID:\n    PMC3759846   
371                               NaN   
1796                PMCID: PMC3599165   
821                        PMC3404461   
820                        PMC3763374   
819                    PMID: 24035434   
818                        PMC3763375   
843                PMCID: PMC3740234    
822                          23117109   
823                        PMC3757156   
824                        PMC3611597   
825                               NaN   
328                        PMC3786614   
243                        PMC3716626   
1837                         23562481   
826                 PMCID: PMC3791421   
225                        PMC3190389   
827                          23201205   
828                      PUB AUG 2013   
829             23200744  PMC3552157    
372                    PMC3472342\n\n   
2106                PMCID: PMC3600532   
986                               NaN   

                                              Publisher  \
439                                            ELSEVIER   
21                            AMERICAN CHEMICAL SOCIETY   
20                            AMERICAN CHEMICAL SOCIETY   
19                            AMERICAN CHEMICAL SOCIETY   
9          ACS (AMERCIAN CHEMICAL SOCIETY) PUBLICATIONS   
8          ACS (AMERCIAN CHEMICAL SOCIETY) PUBLICATIONS   
22                            AMERICAN CHEMICAL SOCIETY   
34               AMERICAN CHEMICAL SOCIETY PUBLICATIONS   
23                            AMERICAN CHEMICAL SOCIETY   
927             INTERNATIONAL UNION OF CRYSTALLOGRAPHY    
922              INTERNATIONAL UNION OF CRYSTALLOGRAPHY   
921              INTERNATIONAL UNION OF CRYSTALLOGRAPHY   
920              INTERNATIONAL UNION OF CRYSTALLOGRAPHY   
923              INTERNATIONAL UNION OF CRYSTALLOGRAPHY   
928       INTERNATIONAL UNION OF CRYSTALLOGRAPHY (IUCR)   
929       INTERNATIONAL UNION OF CRYSTALLOGRAPHY (IUCR)   
1711  SOCIETY FOR PUBLICATION OF ACTA DERMATO-VENERE...   
1716                                           SPRINGER   
924              INTERNATIONAL UNION OF CRYSTALLOGRAPHY   
1717                                           SPRINGER   
1719                                           SPRINGER   
1720                                           SPRINGER   
1718                                           SPRINGER   
2051                                    WILEY-BLACKWELL   
1871     

In [63]:
## Converting whole DataFrame to uppercase in the process
df= df.applymap(lambda x: x.upper() if type(x) == str else x )

In [74]:
## Standardize Entries

df['Journal title'] = df['Journal title'].replace(['PLOSONE', 'PLOS ONE', 'PLOS 1', 'PLOS'], 'PLOS ONE') 
df['Journal title'] = df['Journal title'].replace(['ACTA D', 'ACTA CRYSTALLOGRAPHICA SECTION D', 'ACTA CRYSTALLOGRAPHY D', 'ACTA CRYSTALLOGRAPHICA, SECTION D', 'ACTA CRYSTALLOGRAPHICA SECTION D, BIOLOGICAL CRYSTALLOGRAPHY', 'ACTA CRYSTALLOGRAPHICA SECTION D,  BIOLOGICAL CRYSTALLOGRAPHY'], 'ACTA CRYSTALLOGRAPHICA SECTION D: BIOLOGICAL CRYSTALLOGRAPHY') 
df['Journal title'] = df['Journal title'].replace(['AMERICAN JNL EPIDEMIOLOGY'], 'AMERICAN JOURNAL OF EPIDEMIOLOGY') 
df['Journal title'] = df['Journal title'].replace(['AMERICAN JOURNAL OF MEDICAL GENETICS PART A'], 'AMERICAN JOURNAL OF MEDICAL GENETICS')
df['Journal title'] = df['Journal title'].replace(['ANTIMICROBIAL AGENTS AND CHEMOTHERAPY', 'ANTIMICROBIAL AGFENTS AND CHEMOTHERAPY'], 'ANTIMICROBIAL AGENTS & CHEMOTHERAPY') 
df['Journal title'] = df['Journal title'].replace(['ANGEWANDE CHEMIE', 'ANGEWANDTE CHEMIE INTERNATIONAL EDITION','ANGEW CHEMS INT ED' ], 'ANGEWANDTE CHEMIE')
df['Journal title'] = df['Journal title'].replace(['BEHAVIOUR RESEARCH AND THERAPY'], 'BEHAVIOR RESEARCH & THERAPY')
df['Journal title'] = df['Journal title'].replace(['VIROLOGY JOURNAL'], 'VIROLOGY')

In [80]:
## View final counts of each Journal Entry (as typed)
print(df['Journal title'].value_counts())

PLOS ONE                                                            210
JOURNAL OF BIOLOGICAL CHEMISTRY                                      53
NEUROIMAGE                                                           29
PLOS GENETICS                                                        24
PLOS PATHOGENS                                                       24
NUCLEIC ACIDS RESEARCH                                               23
PLOS NEGLECTED TROPICAL DISEASES                                     20
PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES                      20
HUMAN MOLECULAR GENETICS                                             19
NATURE COMMUNICATIONS                                                19
BMC PUBLIC HEALTH                                                    14
BRAIN                                                                14
MOVEMENT DISORDERS                                                   13
JOURNAL OF NEUROSCIENCE                                         

**The five most common jourals (and the respective number of articles) are: PLOS One (210), Journal of Biological Chemistry (53), Neuroimage (29), PLOS Pathogens (24), and PLOS Genetics (24).**

## Next, calculate the mean, median, and standard deviation of the open-access cost per article for each journal

In [82]:
### Cleaning Article Cost Entries 

In [83]:
## removed currency symbols from cost column

## removed £
df= df.apply(lambda x: x.str.replace('£', ''))

## removed $
df= df.apply(lambda x: x.str.replace('$', ''))

In [89]:
## Replace 999999.00 value with mean

df= df.apply(lambda x: x.str.replace('999999.00', '2015.04'))

In [98]:
## Creating Separate DataFrames for each Journal Title

plos_one_isolated=df.loc[lambda df: df['Journal title'] == 'PLOS ONE', :]

jbc_isolated=df.loc[lambda df: df['Journal title'] == 'JOURNAL OF BIOLOGICAL CHEMISTRY', :]

neuroimage_isolated=df.loc[lambda df: df['Journal title'] == 'NEUROIMAGE', :]

plos_pathogens_isolated=df.loc[lambda df: df['Journal title'] == 'PLOS PATHOGENS', :]

plos_genetics_isolated=df.loc[lambda df: df['Journal title'] == 'PLOS GENETICS', :]

In [99]:
## Calculating Mean, Median, Standard deviation for PLOS One article Costs

## Describe PLOS One costs
plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count       210.000000
mean       1891.382286
std       13229.208992
min         122.310000
25%         853.320000
50%         901.810000
75%        1044.737500
max      192645.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

In [100]:
## Remove Extreme Costs 

r = plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).quantile(0.99)

plos_one_isolated_reduced = plos_one_isolated[plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric) < r]

In [101]:
## Describe Article Costs

plos_one_isolated_reduced['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count     201.000000
mean      937.437612
std       193.968371
min       122.310000
25%       852.430000
50%       897.610000
75%      1040.700000
max      1785.360000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS One Article Cost Statistics <br>
mean:      937.437612 <br>
median:    897.610000 <br>
std:       193.968371**

In [102]:
jbc_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      53.000000
mean     1434.747925
std       415.984485
min       265.670000
25%      1152.720000
50%      1314.530000
75%      1610.310000
max      2501.070000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**Journal of Biological Chemistry Cost Statistics <br>
mean:     1434.747925 <br>
median:      1314.530000 <br>
std:       415.984485**

In [103]:
neuroimage_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      29.000000
mean     2215.168276
std       266.653947
min      1747.160000
25%      2100.540000
50%      2326.430000
75%      2396.570000
max      2503.340000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**Neuroimage Cost Statistics <br>
mean:     2215.168276 <br>
median:      2326.430000 <br>
std:       266.653947**

In [105]:
plos_pathogens_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      24.000000
mean     1609.714583
std       198.700369
min      1254.020000
25%      1440.000000
50%      1600.520000
75%      1760.310000
max      2015.040000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS Pathogens Cost Statistics <br>
mean:     1609.714583 <br>
median:      1600.520000 <br>
std:       198.700369**

In [106]:
plos_genetics_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      24.000000
mean     1674.105000
std       180.284357
min      1394.050000
25%      1484.862500
50%      1718.390000
75%      1778.225000
max      2015.040000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS Genetics Cost Statistics <br>
mean:     1674.105000 <br>
median:      1718.390000 <br>
std:       180.284357**